### PR TITLE
*: fix config reload and recreate client when load topology failed

### PR DESCRIPTION
**Pull request by crazycs520**

### CODE REVIEW

File: component/topology/client.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	c.pdCli = pdCli
+	c.etcdCli = etcdCli
+	log.Info("recreate pd/etcd client success")
+}
+
+func (c *Client) Close() error {
+	return c.etcdCli.Close()
+}
+
+func createClient(cfg *config.Config) (*pdclient.APIClient, *clientv3.Client, error) {
```
Comment (breeswish) [inline component/topology/client.go:53]: Why not move this to TiDB Dashboard? This kind of logic seems to be something common.

File: component/topology/topology.go [hunk 2]
```diff
@@ -34,16 +34,10 @@ func GetCurrentComponent() []Component {
 }
 
 func GetEtcdClient() *clientv3.Client {
-	if discover == nil {
-		return nil
-	}
-	return discover.etcdCli
+	return discover.cli.etcdCli
```
Comment (breeswish) [inline component/topology/topology.go:37]: Please don't do this. Avoid shared globals.

Commonly, for shared globals, you will make these things worse:
1. Now it's hard to write unit tests. Not to mention running unit tests concurrently.
2. Hard to ensure concurrent safety. You left this possible to be modified and accessed EVERYWHERE in the code. This is horrible. You may need to take a look at every references to ensure that it is not modified concurrently.

File: component/topology/topology.go [hunk 3]
```diff
@@ -34,16 +34,10 @@ func GetCurrentComponent() []Component {
...
 }
 
 func Subscribe() Subscriber {
-	if discover == nil {
-		return nil
-	}
 	return discover.Subscribe()
```
Comment (breeswish) [inline component/topology/topology.go:41]: Avoid globals.

File: main.go [hunk 4]
```diff
@@ -68,7 +68,7 @@ func main() {
 	}
 	defer topology.Stop()
 
-	pdvariable.Init(topology.GetEtcdClient())
+	pdvariable.Init(topology.GetEtcdClient)
```
Comment (breeswish) [inline main.go:71]: Avoid globals.

File: main.go [hunk 5]
```diff
@@ -68,7 +68,7 @@ func main() {
...
 	defer pdvariable.Stop()
```
Comment (breeswish) [inline main.go:72]: Avoid globals.

File: main.go [hunk 6]
```diff
@@ -68,7 +68,7 @@ func main() {
...
 
 	topsql.Init(document.Get(), timeseries.InsertHandler, timeseries.SelectHandler, topology.Subscribe())
```
Comment (breeswish) [inline main.go:74]: Avoid globals.

File: component/topology/client.go (hunk 1 above)
Comment (crazycs520) [reply component/topology/client.go:53]: em... It is not common now...

File: component/topology/topology.go (hunk 2 above)
Comment (crazycs520) [reply component/topology/topology.go:37]: Ok, I will do some refactor in another PR, This PR is a bug fix, and there are many other global share variable.

**Pull request merged**